src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -558,10 +567,64 @@ const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
   const onSuccessEditingName = useCallback(() => {
     setTimeout(() => dispatch(fetchSortedFolderContentThunk(currentFolderId)), 500);
   }, [currentFolderId]);
+
   const onCloseEditNameDialog = useCallback(() => {
     setEditNameItem(null);
   }, []);
 
+  const onCloseStopSharingDialog = useCallback(() => {
+    setShowStopSharingConfirmation(false);
+  }, []);
+
+  const onOpenStopSharingAndMoveToTrashDialog = useCallback(() => {
+    setShowStopSharingConfirmation(true);
+  }, []);
+
+  const onStopSharingAndMoveToTrash = async () => {
+    const items = selectedItems;
+
+    setIsStopSharingDialogLoading(true);
+
+    const stopSharingItems = items.map(async (item) => {
+      let itemName: string;
+      const isSharedItem = item.sharings && item.sharings.length > 0;
+
+      if (isSharedItem) {
+        if (item.name.length > MAX_SHARED_NAME_LENGTH) {
+          itemName = item.name.substring(0, 32).concat('...');
+        } else {
+          itemName = item.name;
+        }
+
+        await dispatch(
+          sharedThunks.stopSharingItem({
+            itemType: item.isFolder ? 'folder' : 'file',
+            itemId: item.uuid as string,
+            itemName,
+          }),
+        );
+
+        return item;
+      }
+    });
+
+    await Promise.all(stopSharingItems);
+
+    const itemsToTrash = items.map((selectedShareItem) => ({
+      ...selectedShareItem,
+      isFolder: selectedShareItem.isFolder,
+    }));
+
+    await moveItemsToTrash(itemsToTrash);
+
+    setTimeout(async () => {
+      dispatch(fetchSortedFolderContentThunk(currentFolderId));
+    }, 500);
+
+    setShowStopSharingConfirmation(false);
+    setIsStopSharingDialogLoading(false);
+  };
+
```

Comment:
Try to move all the logic of `StopSharingAndMoveToTrashDialog` out of drive explorer, for example can create a wrapper to add this logic.